stackslib/src/net/relay.rs

```diff
@@ -528,6 +529,41 @@ impl Relayer {
         Ok(())
     }
 
+    /// Given Nakamoto blocks pushed to us, verify that they correspond to expected block data.
+    pub fn validate_nakamoto_blocks_push(
+        conn: &SortitionDBConn,
+        blocks_data: &NakamotoBlocksData,
+    ) -> Result<(), net_error> {
+        for NakamotoBlocksDatum(consensus_hash, block) in blocks_data.blocks.iter() {
+            let block_hash = block.header.block_hash();
+
+            // is this the right Stacks block for this sortition?
+            let sn = match SortitionDB::get_block_snapshot_consensus(conn.conn(), consensus_hash)? {
```

Comment:
This should be loading the sortition using `block.header.consensus_hash`

Reply:
I mentioned this below, but I'm not sure that using `block.header.consensus_hash` works here...

`winning_stacks_block_hash` in Nakamoto will correspond to a Stacks block previously mined -- so its block header's `consensus_hash` field points at a different bitcoin block. Instead, it needs to be looked up by the `announcement_ch` (or the like: the consensus hash in which the winning miner announced the block hash).

Reply:
Otherwise, the checks here need to all be made against the Tenures DB rather than SortitionDB.